torch/csrc/jit/codegen/cuda/lower_scalar_hoist.cpp

```diff
@@ -220,21 +220,44 @@ std::vector<Val*> CommonIndexMap::getHoistedIndices(kir::ForLoop* loop) const {
 
 namespace {
 
+// Check if the given `value` is already allocated (and computed in full) or
+// computed partially. If yes, then return the position of the existing
+// computation for the first return value, else return -1 for the first return
+// value. The second return value is a boolean indicating if the given `value`
+// is already fully computed.
+std::pair<int64_t, bool> findAllocPointFromDataDependency(
```

Comment:
As I start hoisting more scalar expressions, I need a smarter way to determine the place to insert allocation. I also need to know if a subexpression is already allocated, and if yes, don't insert new allocation.